### ISSUE TITLE
Dissect plugin prase doesn't correct with empty field in content

### ISSUE THREAD

**Issue by qianghaohao** [logstash-plugins/logstash-filter-dissect#41]
@ph @joshuar @colinsurprenant @elasticdog @karmi 
I use the dissect plugin with logstash to parse the log which has empty fileds, but is doesn't match fields correctly.The folowing is my log pattern and logstash config file and logstash output:
**log pattern:**
```2017-10-24|||500|```
**logstash config file:**
```
input {
  beats {
    port => 5044
  }
}

filter {
   ruby {
        code => "
            event.timestamp.time.localtime
            tstamp = event.get('@timestamp').to_i
            event.set('date_str', Time.at(tstamp).strftime('%Y-%m-%d'))
        "
   }

   dissect {
          mapping => {
              "message" => "%{time_local}|%{server_ip}|%{request}|%{status_code}|%{upstrame}"
          }
  }
}
output {
    stdout { codec => rubydebug }
}
```
**logstash parse result:**
```
{
       "date_str" => "2017-10-23",
        "request" => "",
    "status_code" => "",
       "business" => "nginx",
         "offset" => 217,
     "input_type" => "log",
     "time_local" => "2017-10-24",
         "source" => "/home/ec2-user/filebeat/test.log",
        "message" => "2017-10-24|||500|",
           "type" => "access",
           "tags" => [
        [0] "beats_input_codec_plain_applied"
    ],
     "@timestamp" => 2017-10-24T04:00:22.130Z,
       "@version" => "1",
           "beat" => {
        "hostname" => "awsuw7-50.opi.com",
            "name" => "awsuw7-50.opi.com",
         "version" => "5.5.1"
    },
           "host" => "awsuw7-50.opi.com",
      "server_ip" => "500",
       "upstrame" => ""
}
```
So, as we have seen, the parse was is not correct, the server_ip filed should be "", request filed shuld be "", status_code filed shuld be 500, but the result is not what i want. So, what's the reason, how to fix this problem?
- Version: 5.5.1
- Operating System: CentOs 7

**Comment by qianghaohao** [comment]
Is this a bug?

**Comment by guyboertje** [comment]
@qianghaohao 
Yes it is - in the current plugin. Sorry about that. Its probably the only defect that stops Dissect from being used more widely.

Its fixed in master but the plugin has not been released. There is a documentation change PR pending - as soon as that is merged then I will release a new version.

This is the original bug issue  https://github.com/logstash-plugins/logstash-filter-dissect/issues/11

**Comment by qianghaohao** [comment]
@guyboertje Very good！So, What time can newest version  be released? Very much looking forward to.

**Comment by qianghaohao** [comment]
I am wating for this bufix for several days.

**Comment by qianghaohao** [comment]
@guyboertje Hello，this bug is fixed in version 1.1.1?
